### PR TITLE
Alerting: Add validation to slack contact point

### LINKED ISSUE #44414: Alerting: Slack contact point is missing front-end validation
**What happened**:

The Slack contact point allows configuring a `webhook URL` or a `Recipient` and `Token`.

The front-end is currently not validating this.

image

**What you expected to happen**:

Not filling in any details about the Slack contact point should show a front-end error.

**How to reproduce it (as minimally and precisely as possible)**:

1. Create a new Contact Point
2. Choose Slack
3. Save without filling in any details

**Anything else we need to know?**:

**Environment**:
- Grafana version:
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by peterholmberg**
**What this PR does / why we need it**:
Adds frontend validation to the slack contact point. Slack needs a `token` or a `webhook url` but not both. Therefore we need to change what is required depending on if either webhook or token is set. If token is set we also need a recipient (channel). 

**Which issue(s) this PR fixes**:
Fixes #44414

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/alerting/unified/components/receivers/form/fields/OptionField.tsx [hunk 1]
```diff
@@ -157,10 +157,27 @@ const OptionInput: FC<Props & { id: string }> = ({ option, invalid, id, pathPref
...
+    return option.required ? 'Required' : false;
+  }
+
+  const dependentOn = getValues(`items[0].${option.dependsOn}`);
+  return !dependentOn && option.required ? 'Required' : false;
+};
+
+const determineReadOnly = (option: NotificationChannelOption, getValues: any) => {
+  if (!option.dependsOn) {
+    return false;
```
Comment (konrad147) [inline public/app/features/alerting/unified/components/receivers/form/fields/OptionField.tsx:179]: Why always false?

File: pkg/tests/api/alerting/api_available_channel_test.go [hunk 2]
```diff
@@ -72,7 +72,8 @@ var expAvailableChannelJsonOutput = `
         },
         "required": true,
         "validationRule": "",
-        "secure": false
+        "secure": false,
```
Comment (alexweav) [inline pkg/tests/api/alerting/api_available_channel_test.go:75]: Just fyi - there's an [open PR](https://github.com/grafana/grafana/pull/45751) which deletes this entire blob so it doesn't have to be hand updated every time the declarative code in `available_channels.go` changes.

Comment (alexweav) [reply pkg/tests/api/alerting/api_available_channel_test.go:75]: If you don't want to deal with all this, that one will be going in soon.

Comment (peterholmberg) [reply pkg/tests/api/alerting/api_available_channel_test.go:75]: Praise!

File: public/app/features/alerting/unified/components/receivers/form/fields/OptionField.tsx (hunk 1 above)
Comment (konrad147) [reply public/app/features/alerting/unified/components/receivers/form/fields/OptionField.tsx:179]: My fault. This is correct :)

**Review by konrad147 — APPROVED** [review]
Nice! 🚀

**Pull request merged**